ovn_k8s/modes/overlay.py

```diff
@@ -285,21 +286,14 @@ def add_endpoint(self, event):
         endpoint_data = event.metadata
         service_name = endpoint_data['metadata']['name']
         namespace = endpoint_data['metadata']['namespace']
+        ips = endpoint_data.get('custom', {}).get('ips')
 
         vlog.dbg("received endpoint data %s" % (endpoint_data))
 
-        ips = []
-        subsets = endpoint_data.get('subsets')
-        if subsets:
-            for subset in subsets:
-                addresses = subset.get('addresses')
-                if not addresses:
-                    continue
-
-                for address in addresses:
-                    ip = address.get('ip')
-                    if ip:
-                        ips.append(ip)
+        if not ips:
+            vlog.dbg("No IP addresses for endpoint %s, no need to update VIP" %
+                     service_name)
```

Comment:
When "ips" is None, we still want to update the vips. For e.g., let us say you have a service with 3 endpoints. You delete the endpoints one after the other. Your last endpoint update will come with no "ips". If you return without doing anything, you will have a stale load balancer entry remaining in the database.